Makefile

```diff
@@ -1,27 +1,60 @@
 .PHONY: all container push clean node-problem-detector
 
-all: push
+all: container
 
 # See node-problem-detector.yaml for the version currently running-- bump this ahead before rebuilding!
-TAG = v0.2
-
-PROJ = google_containers
+# TAG is the tag of the container image.
+TAG?=v0.3
+
+# PROJ is the image project.
+PROJ?=google_containers
+
+# ENABLE_JOURNALD enables build journald support or not. Building journald support needs libsystemd-dev
+# or libsystemd-journal-dev.
+# TODO(random-liu): Build NPD inside container.
+ENABLE_JOURNALD?=1
+
+# TODO(random-liu): Support different architectures.
+BASEIMAGE:=alpine:3.4
+
+# Disable cgo by default to make the binary statically linked.
+CGO_ENABLED:=0
+
+# NOTE that enable journald will increase the image size.
+ifeq ($(ENABLE_JOURNALD), 1)
+	# Enable journald build tag.
+	BUILD_TAGS:=-tags journald
+	# Use fedora because it has newer systemd version (229) and support +LZ4. +LZ4 is needed
+	# on some os distros such as GCI.
+	BASEIMAGE:=fedora
+	# Enable cgo because sdjournal needs cgo to compile. The binary will be dynamically
+	# linked if CGO_ENABLED is enabled. This is fine because fedora already has necessary
+	# dynamic library. If we need to make this statically linked in the future, add -ldflags
+	# '-extldflags "-static"'
+	CGO_ENABLED:=1
+endif
 
 PKG_SOURCES := $(shell find pkg -name '*.go')
 
 node-problem-detector: ./bin/node-problem-detector
 
 ./bin/node-problem-detector: $(PKG_SOURCES) node_problem_detector.go
-	CGO_ENABLED=0 GOOS=linux godep go build -a -installsuffix cgo -ldflags '-w' -o node-problem-detector
+	CGO_ENABLED=$(CGO_ENABLED) GOOS=linux go build -o ./bin/node-problem-detector -a -ldflags '-w' $(BUILD_TAGS)
 
-test:
-	go test -timeout=1m -v -race ./pkg/...
+Dockerfile: Dockerfile.in
```

Comment:
If you don't add Dockerfile to PHONY targets, users will be confused. Consider this scenario:
```bash
make
# The Dockerfile file didn't exist and make created it.
# @BASEIMAGE@ is Fedora and systemd-journal is enabled.
make ENABLE_JOURNALD=0
# The Dockerfile file already exists and make skipped the Dockerfile target. 
# However the user believes that @BASEIMAGE@ is set to "alpine:3.4" and
# systemd-journal is disabled.
```